core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala

```diff
@@ -81,6 +84,24 @@ class DeltaLog private(
 
   protected def spark = SparkSession.active
 
+  /**
+   * Issue 1144 - Verify the required Spark conf for delta
+   * Throw `DeltaErrors.configureSparkSessionWithExtensionAndCatalog` exception if both
+   * `spark.sql.extensions` and `spark.sql.catalog.spark_catalog` config are missing.
+   * If `DeltaSparkSessionExtension` is alternatively activated using the `.withExtension()` API,
+   * we expect the DeltaSqlParser to be available and try parse the vacuum command.
```

Comment:
> try parse the vacuum command.

Why do we need to try to parse the command?

Reply:
We also have a case where a SparkSession is initialized with `.withExtensions()` API, instead of using the `.config("spark.sql.extensions", "")` api in which case `spark.sparkContext.getConf` won't be able to find `spark.sql.extensions` config. `DeltaSQLCommandTest` also uses this pattern. This is why I also had to try parsing a delta SQL command to see if the Delta SQL parser is available. We still want to do this right?

Reply:
Ah, I didn't know `withExtensions`. How about just checking  the `V2_SESSION_CATALOG_IMPLEMENTATION` config? I haven't seen any one who sets `V2_SESSION_CATALOG_IMPLEMENTATION` but forgets `spark.sql.extensions`.

Reply:
sounds good. Will just check for `V2_SESSION_CATALOG_IMPLEMENTATION` config.